### ISSUE TITLE
Building vApp fails

### ISSUE THREAD

**Issue by vchrisb** [frapposelli/vagrant-vcloudair#5]
I'm not able to provision to vcloud air.
In stage "Building vApp..." I'm receiving following error:

```
.vagrant.d/gems/gems/vagrant-vcloudair-0.5.2/lib/vagrant-vcloudair/action/build_vapp.rb:117:in `call': undefined method `last' for nil:NilClass (NoMethodError)
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
...
```
- Vagrant 1.7.2
- vagrant-vcloudair (0.5.2)
- Windows 7

**Comment by tsugliani** [comment]
Can you provide your Vagrantfile configuration ? (clean the password, and confidential fields if any)
Then launch the following command:

`VAGRANT_LOG=debug vagrant up`

**Comment by vchrisb** [comment]
sure:

```
Vagrant.configure(2) do |config|

  config.vm.box = "vchrisb/openSUSE-13.2_64"

  # Set up hostname
  config.vm.hostname = "test.lab.local"

  # vCloud Air provider settings
  config.vm.provider :vcloudair do |vcloudair|

    vcloudair.username = ''
    vcloudair.password = ''

    # if you're using a vCloud Air Dedicated Cloud, put the cloud id here, if
    # you're using a Virtual Private Cloud, skip this parameter.
    vcloudair.vdc_name = ''

    # Set the network to deploy our VM on (case sensitive)
    vcloudair.vdc_network_name = ''

    # Set our Edge Gateway and the public IP we're going to use.
    vcloudair.vdc_edge_gateway = ''
    vcloudair.vdc_edge_gateway_ip = ''

    # Catalog that holds our templates.
    vcloudair.catalog_name = 'Vagrant'

    # Set our Memory and CPU to a sensible value for Docker.
    vcloudair.memory = 1024
    vcloudair.cpus = 2
  end

end
2
```

The debug log is quite lengthy and does include a lot of sensitive date. Here is the output of the last lines:

```
 INFO inventory_check: Catalog [Vagrant] exists
 INFO inventory_check: Using catalog item [vchrisb/openSUSE-13.2_64] in Catalog [Vagrant]...
 INFO interface: info: Testing Network Configuration in vCloud Air...
 INFO interface: info: ==> default: Testing Network Configuration in vCloud Air...
DEBUG driver_5_1: Our IP range: 391968161..391968161
DEBUG driver_5_1: Our Test IP 391962948
DEBUG driver_5_1: Our IP range: 391962948..391962948
DEBUG driver_5_1: Our Test IP 391962948
DEBUG driver_5_1: Found IP 23.92.225.68
DEBUG driver_5_1: Our IP range: 391962944..391962944
DEBUG driver_5_1: Our Test IP 391962948
INFO warden: Calling IN action: #<Vagrant::Action::Builtin::Call:0x40f6308>
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 1 hooks defined.
 INFO runner: Running action: #<Vagrant::Action::Builder:0x3bb2948>
 INFO warden: Calling IN action: #<VagrantPlugins::VCloudAir::Action::IsCreated:0x3bbbfa8>
 WARN is_created: vApp has not been created
 INFO warden: Calling OUT action: #<VagrantPlugins::VCloudAir::Action::IsCreated:0x3bbbfa8>
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 1 hooks defined.
 INFO runner: Running action: #<Vagrant::Action::Warden:0x2c83a10>
 INFO warden: Calling IN action: #<Proc:0x37ef138@C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:94 (lambda)>
 INFO warden: Calling IN action: #<VagrantPlugins::VCloudAir::Action::BuildVApp:0x2c83968>
DEBUG build_vapp: DNS1: 8.8.8.8 DNS2: 8.8.4.4
 INFO interface: info: Building vApp...
 INFO interface: info: ==> default: Building vApp...
ERROR warden: Error occurred: undefined method `last' for nil:NilClass
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: undefined method `last' for nil:NilClass
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: undefined method `last' for nil:NilClass
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x40f6308>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x408a1e8>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: undefined method `last' for nil:NilClass
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: undefined method `last' for nil:NilClass
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO environment: Released process lock: machine-action-df6fca0dd86845707f9ef793863f9bd8
 INFO environment: Running hook: environment_unload
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 1 hooks defined.
 INFO runner: Running action: #<Vagrant::Action::Builder:0x38bfbc0>
C:/Users/USER/.vagrant.d/gems/gems/vagrant-vcloudair-0.5.2/lib/vagrant-vcloudair/action/build_vapp.rb:117:in `call': undefined method `last' for nil:NilClass (NoMethodError)
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/util/busy.rb:19:in `busy'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/call.rb:53:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/Users/USER/.vagrant.d/gems/gems/vagrant-vcloudair-0.5.2/lib/vagrant-vcloudair/action/inventory_check.rb:15:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/Users/USER/.vagrant.d/gems/gems/vagrant-vcloudair-0.5.2/lib/vagrant-vcloudair/action/connect_vcloud.rb:49:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/util/busy.rb:19:in `busy'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/call.rb:53:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/util/busy.rb:19:in `busy'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `run'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:214:in `action_raw'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:191:in `block in action'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:516:in `lock'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:178:in `call'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:178:in `action'
    from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

I get the same error on an Ubuntu 14.04.

Thank You!

**Comment by rustic** [comment]
I am getting the same error when I try to provision to vCloud Air. It happens after creating the vApp Template and also when the template already exists:

vagrant (1.7.4)
vagrant-vcloudair (0.5.2)
Fedora 22

```
Bringing machine 'vchs-oc01' up with 'vcloudair' provider...
==> vchs-oc01: Adding VM to existing vApp...
/home/builduser/.vagrant.d/gems/gems/vagrant-vcloudair-0.5.2/lib/vagrant-vcloudair/action/build_vapp.rb:182:in `call': undefined method `last' for nil:NilClass (NoMethodError)
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
...
```

Here is my existing Vagrantfile:

``` ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :
ENV['VAGRANT_DEFAULT_PROVIDER'] = 'vcloudair'

nodes = [
  { hostname: 'vchs-oc01', box: 'centos7-x64-vmware_ovf-1' },
]

Vagrant.configure('2') do |config|

  # vCloud Air provider settings
  config.vm.provider :vcloudair do |vcloudair|

    vcloudair.username = '********'
    vcloudair.password = '********'

    vcloudair.vdc_name = '********'

    vcloudair.network_bridge = true
    vcloudair.vdc_network_name = 'VCHS Server Net'
    vcloudair.catalog_name = 'Vagrant'

    vcloudair.memory = 2048
    vcloudair.cpus = 2
  end

  nodes.each do |node|
    config.vm.define node[:hostname] do |node_config|
      node_config.vm.box = node[:box]
      node_config.vm.hostname = node[:hostname]
    end
  end

end
```
